### PR TITLE
DOC: Edited percentile docstring

### PR DESCRIPTION
**Pull request by mikep2016**
Added additional methods for percentile estimation and their use in popular software.

### CODE REVIEW

File: numpy/lib/function_base.py [hunk 1]
```diff
@@ -3900,7 +3900,41 @@ def percentile(a, q, axis=None, out=None,
     array([ 7.,  2.])
     >>> assert not np.all(a == b)
 
+    Selected methods for percentile estimation and their use in popular
```
Comment (matthew-brett) [inline numpy/lib/function_base.py]: These should go in the Notes section above.  I'd remove the section underline and just have a bulleted list, preceding each method with an asterisk, as in:

```
* Method A: p(n+1): cited for example by Altman1 and CLSI2 ...
```

File: numpy/lib/function_base.py [hunk 2]
```diff
@@ -3900,7 +3900,41 @@ def percentile(a, q, axis=None, out=None,
...
+    software.
+    --------
+    Method A: p(n+1)
+        Method A, cited for example by Altman1
```
Comment (matthew-brett) [inline numpy/lib/function_base.py]: References for `Altman1`, `CLSI2`?

**Comment by matthew-brett** [comment]
Thanks a lot for taking the time to do this.

**Comment by charris** [comment]
Closes #7875.

**Comment by juliantaylor** [comment]
I don't think is very useful to document percentile methods that we do not implement in the docstring.
It may be sufficient to just cite the percentile paper the R page references and say which if the methods are implemented under which name in numpy. (I think we only have one of them)

**Comment by charris** [comment]
@mikep2016 Please update this.

File: numpy/lib/function_base.py [hunk 3]
```diff
@@ -3900,7 +3900,41 @@ def percentile(a, q, axis=None, out=None,
...
+    Method B: 0.5+pn
+        Method B, cited for example by Armitage et al. and Lentner calculates
+        an index 0.5+pn, and is attributed to Hazen Method B is used in the
+        software programs Mathematica (Wolfram Research Inc., Champaign, IL)
+        and MedCalc (MedCalc Software bvba, Mariakerke, Belgium)
+        (only method, except for reference interval estimation)
+
+    Method C: p(n-1)+1
+        Method C uses p(n-1)+1 and is attributed to Gumbel.4
+        This method is implemented in both Microsoft Excel 200318 and 2007
```
Comment (bashtage) [inline numpy/lib/function_base.py]: Should this be 200318, or 2003?

Comment (charris) [inline numpy/lib/function_base.py]: Google seems to indicate that `2003` would be correct.

File: numpy/lib/function_base.py (hunk 1 above)
Comment (charris) [inline numpy/lib/function_base.py]: Agree, Also
```
    software.
    --------
```
Is not a recogized heading.

File: numpy/lib/function_base.py [hunk 4]
```diff
@@ -3900,7 +3900,41 @@ def percentile(a, q, axis=None, out=None,
...
+        (Microsoft Corp., Redmond, WA)19 as well as in the Calc program,
+        part of the OpenOffice suite by Sun Microsystems (Oracle Corp., Redwood
+        Shores, CA), although this does not seem to be documented. Method C is
+        the default value in R20 and is also among the methods available in
+        Statistica(StatSoft, Inc., Tulsa, OK)
+
+    Method D: p(n+1/3)+1/3
+        Method D uses p(n+1/3)+1/3 and is proposed by Hyndman and Fan as the
+        best choice among many othermethods including the above. We are not
```
Comment (matthew-brett) [inline numpy/lib/function_base.py]: `othermethods` should be `other methods`.

**Comment by juliantaylor** [comment]
I'm still -1 on putting information into the docstring that has no relation to the actual function.
I'll close this for now. But just putting in a link to the paper is fine with me, feel free to reopen then.

**Comment by matthew-brett** [comment]
I think it would be a shame to lose this information.  It arose, as I remember, because of confusion about the result of `percentile` compared to other software.  If you don't want that information in the docstring, can you suggest somewhere else it could go?

**Comment by juliantaylor** [comment]
The information is not lost, it is in the R documentation and in a published paper. It just has little relation with what numpy implements so I don't think we need it in our documentation.
Having the information linked more is useful, adding a reference with a note we don't implement it should be fine.

**Comment by apbard** [comment]
For what is worth, I agree with @juliantaylor. As I wrote in the original [discussion]( https://github.com/numpy/numpy/issues/7875#issuecomment-267793576), just one of the [Hyndman-Fan (H&F)](https://www.amherst.edu/media/view/129116/original/Sample+Quantiles.pdf) methods is implemented  in numpy. Also, as I tried to explain in that post, IMHO the names current implementation names are quite misleading and could be revised or integrated with some of the H&F methods.

**Comment by matthew-brett** [comment]
Sorry by 'lost' I don't mean lost to research, but relatively inaccessible to someone using numpy.  If the R docstring is easy to link to and to read, and it's clear how to relate it to the arguments etc in the numpy function, that's fine.

**Pull request closed without merging**